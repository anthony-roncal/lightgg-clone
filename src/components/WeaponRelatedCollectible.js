import { React } from 'react';

const WeaponRelatedCollectible = ({ icon, name, source }) => {
    return (
        <>
            <h3 className='section-header'>Related Collectible</h3>
            <div className='weapon-related-collectible'>
                <img src={icon} alt='icon' />
                <div className='weapon-related-collectible-text'>
                    <h1><a href="#">{name}</a></h1>
                    <p className='weapon-source'>{source}</p>
                </div>
            </div>
        </>
    );
}

export default WeaponRelatedCollectible;