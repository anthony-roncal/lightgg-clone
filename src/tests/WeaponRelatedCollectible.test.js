import React from "react";
import renderer from "react-test-renderer";
import WeaponRelatedCollectible from "../components/WeaponRelatedCollectible";

const mockData = {
    "displayProperties": {
        "description": "",
        "name": "Gjallarhorn",
        "icon": "/common/destiny2_content/icons/b62083eed6a4708e581fc9a061bcc8e9.jpg",
        "hasIcon": true
    },
    "collectibleData": {
        "sourceString": "Source: \"And Out Fly the Wolves\" Quest"
    }
};

const BUNGIE_URL = 'https://www.bungie.net';

describe("Weapon Related Collectible component", () => {
    it("renders correct elements and data", () => {
        const tree = renderer.create(
            <WeaponRelatedCollectible
                icon={`${BUNGIE_URL}${mockData.displayProperties.icon}`}
                name={mockData.displayProperties.name}
                source={mockData.collectibleData.sourceString}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
