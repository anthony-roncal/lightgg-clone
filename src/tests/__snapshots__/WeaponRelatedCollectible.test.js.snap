// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weapon Related Collectible component renders correct elements and data 1`] = `
Array [
  <h3
    className="section-header"
  >
    Related Collectible
  </h3>,
  <div
    className="weapon-related-collectible"
  >
    <img
      alt="icon"
      src="https://www.bungie.net/common/destiny2_content/icons/b62083eed6a4708e581fc9a061bcc8e9.jpg"
    />
    <div
      className="weapon-related-collectible-text"
    >
      <h1>
        <a
          href="#"
        >
          Gjallarhorn
        </a>
      </h1>
      <p
        className="weapon-source"
      >
        Source: "And Out Fly the Wolves" Quest
      </p>
    </div>
  </div>,
]
`;
